graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/buffers/MessageEvent.java

```diff
@@ -45,6 +46,7 @@ public MessageEvent newInstance()
 
     private RawMessage raw;
     private Message msg;
```

Comment:
I think that's a disaster waiting to happen. Either `msg` or `msgList` is not null but never both. So you have to know exactly, which one has been set before or always check both.